tsdb/wal/checkpoint.go

```diff
@@ -270,6 +270,16 @@ func Checkpoint(logger log.Logger, w *WAL, from, to int, keep func(id uint64) bo
 	return stats, nil
 }
 
+func GetCheckpointSize(walDir string) (sizes int64) {
+	walDir, _ = filepath.Abs(walDir)
+	dirs, _ := listCheckpointDirs(walDir)
+	for _, dirName := range dirs {
+		size, _ := fileutil.DirSize(fmt.Sprintf("%s/%s", walDir, dirName))
```

Comment:
We should collect these errors in MultiError and return this (while we check all dirs even if a dir in between causes error).